packages/babel-types/src/definitions/utils.js

```diff
@@ -161,6 +161,35 @@ export function assertValueType(type: string): Validator {
   return validate;
 }
 
+export function assertShape(shape: {| [string]: Validator |}): Validator {
```

Comment:
```suggestion
export function assertShape(shape: { [string]: Validator }): Validator {
```

indexers don't work with exact objects.